pallets/base-nft/src/lib.rs

```diff
@@ -231,33 +231,39 @@ impl<T: Config> Pallet<T> {
 
 	/// Transfer NFT(non fungible token) from `from` account to `to` account
 	pub fn transfer(
-		from: &T::AccountId,
-		to: &T::AccountId,
+		from: Vec<&T::AccountId>,
+		to: Vec<&T::AccountId>,
 		token: (T::ClassId, T::TokenId),
+		fraction_amount: u8,
 	) -> DispatchResult {
 		Tokens::<T>::try_mutate(token.0, token.1, |token_info| -> DispatchResult {
 			let mut info = token_info.as_mut().ok_or(Error::<T>::TokenNotFound)?;
-			ensure!(info.owner == *from, Error::<T>::NoPermission);
+			ensure!(
+				info.owners.iter().any(|v| from.iter().any(|w| w == &v)),
+				Error::<T>::NoPermission
+			);
+
 			if from == to {
 				// no change needed
 				return Ok(());
 			}
 
-			info.owner = to.clone();
+			// TODO: assign partial ownership according to fraction_amount
+			info.owners = to.into_iter().cloned().collect();
 
 			#[cfg(not(feature = "disable-tokens-by-owner"))]
 			{
 				TokensByOwner::<T>::remove(from, token);
-				TokensByOwner::<T>::insert(to, token, ());
+				TokensByOwner::<T>::insert(info.owners.clone(), token, ());
 			}
 
 			Ok(())
 		})
 	}
 
-	/// Mint NFT(non fungible token) to `owner`
+	/// Mint NFT(non fungible token) to `owners`
 	pub fn mint(
-		owner: &T::AccountId,
+		owners: Vec<&T::AccountId>,
```

Comment:
I think `mint` should also require a single owner. Later he may want to transfer / sell a part of it